integration/nodejs_stack_integration_test.go

```diff
@@ -0,0 +1,352 @@
+		Expect(docker.Image.Remove.Execute(image.ID)).To(Succeed())
+		Expect(docker.Volume.Remove.Execute(occam.CacheVolumeNames(name))).To(Succeed())
+		Expect(os.RemoveAll(builderConfigFilepath)).To(Succeed())
+	})
+
+	context("When using Node.js 16 as run image", func() {
+		var (
+			err                error
+			builder            string
+			buildImageID       string
+			runImageID         string
+			runNodejs16ImageID string
+			nodeMajorVersion   int
+		)
+
+		it.After(func() {
+			Expect(docker.Image.Remove.Execute(builder)).To(Succeed())
+			Expect(docker.Image.Remove.Execute(buildImageID)).To(Succeed())
+			Expect(docker.Image.Remove.Execute(runImageID)).To(Succeed())
+			Expect(docker.Image.Remove.Execute(runNodejs16ImageID)).To(Succeed())
+			Expect(os.RemoveAll(source)).To(Succeed())
+		})
+
+		it.Before(func() {
+			nodeMajorVersion = 16
+
+			buildImageID, runImageID, builderConfigFilepath, builder, err = generateBuilder(root)
+			Expect(err).NotTo(HaveOccurred())
+
+			source, err = occam.Source(filepath.Join("testdata", "nodejs_simple_app"))
+			Expect(err).NotTo(HaveOccurred())
+
+			//Creating and pushing the run image to local registry
+			runNodejs16Archive := filepath.Join(root, fmt.Sprintf("../build-nodejs-%d", nodeMajorVersion), "run.oci")
+			runNodejs16ImageID = fmt.Sprintf("run-nodejs-%d-%s", nodeMajorVersion, uuid.NewString())
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(archiveToDaemon(runNodejs16Archive, runNodejs16ImageID)).To(Succeed())
+		})
+
+		it("sucessfully builds an app", func() {
+
+			image, _, err = pack.Build.
+				WithExtensions(
+					settings.Extensions.UbiNodejsExtension.Online,
+				).
+				WithBuildpacks(
+					settings.Buildpacks.NodeEngine.Online,
+					settings.Buildpacks.NPMInstall.Online,
+					settings.Buildpacks.BuildPlan.Online,
+				).
+				WithBuilder(builder).
+				WithNetwork("host").
+				WithEnv(map[string]string{"BP_NODE_RUN_EXTENSION": runNodejs16ImageID}).
+				WithPullPolicy("always").
+				Execute(name, source)
+			Expect(err).NotTo(HaveOccurred())
+
+			container, err = docker.Container.Run.
+				WithPublish("8080").
+				WithCommand("node server.js").
+				Execute(image.ID)
+			Expect(err).NotTo(HaveOccurred())
+
+			Eventually(container).Should(Serve("Hello World!"))
+		})
+	})
+
+	context("When using Node.js 18 as run image", func() {
+		var (
+			err                error
+			builder            string
+			buildImageID       string
+			runImageID         string
+			runNodejs18ImageID string
+			nodeMajorVersion   int
+		)
+
+		it.After(func() {
+			Expect(docker.Image.Remove.Execute(builder)).To(Succeed())
+			Expect(docker.Image.Remove.Execute(buildImageID)).To(Succeed())
+			Expect(docker.Image.Remove.Execute(runImageID)).To(Succeed())
+			Expect(docker.Image.Remove.Execute(runNodejs18ImageID)).To(Succeed())
+			Expect(os.RemoveAll(source)).To(Succeed())
+		})
+
+		it.Before(func() {
+			nodeMajorVersion = 18
```

Comment:
Is there any way to have an array of versions defined for example [16, 18, 20] and then iterate over last list to run the tests for each version instead of having to duplicate the code for each version?

Reply:
Seems like it is a thing in go - https://golang.cafe/blog/golang-table-test-example

Reply:
That was the main idea and I would really like to follow this pattern, but it doesnt follow the pattern we have ATM in the buildpack integration tests, so that was the main reason I rejected it. But compared to code reusability, i think is much more preferable. Thaks!